plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -1539,7 +1593,40 @@ def exec_module(self, **kwargs):
                 current_nics = []
                 results = vm_dict
                 current_osdisk = vm_dict['storage_profile']['os_disk']
-                current_ephemeral = current_osdisk.get('diff_disk_ettings', None)
+                current_osdisk_delete = current_osdisk.get('delete_option')
+                if self.os_disk_delete_option and self.os_disk_delete_option != current_osdisk_delete:
+                    self.log('CHANGED: OS disk delete_option differs.')
+                    differences.append('OS Disk delete_option')
+                    changed = True
+
+                # Data disks delete option diff
+                if self.data_disks and vm_dict['storage_profile'].get('data_disks'):
+                    for dd_arg in self.data_disks:
+                        if dd_arg.get('delete_option') is None:
+                            continue
```

Comment:
This needs to be checked in case the remote has set the `delete_option` to Delete.